src/ast.rs

```diff
@@ -104,6 +104,14 @@ pub struct NumericFor<'a> {
     pub body: Chunk<'a>,
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct GenericFor<'a> {
+    #[serde(borrow)]
+    pub vars: Vec<Cow<'a, str>>,
+    pub iterator: Expression<'a>,
```

Comment:
Ah, Lua's generic for is more generic than that. You should check out the grammar.

This is valid, for example:
```lua
for k, v in next, t do
end
```

So perhaps we need `iteration_source: Vec<Expression<'a>>` instead of `iterator: Expression<'a>`?